### ISSUE TITLE
SEGV in MP4Box

### ISSUE THREAD

**Issue by rbouqueau** [gpac/gpac#2649]
poc

https://github.com/gandalf4a/crash_report/blob/main/gpac/MP4Box/poc3/segv_291

./MP4Box -dash 10000 poc

```
Description

new 3 SEGV in MP4Box
Version

$ ./MP4Box -version
MP4Box - GPAC version 2.3-DEV-rev566-g50c2ab06f-master

Platform

$ uname -a
Linux user-GE40-2PC-Dragon-Eyes 6.2.0-33-generic #33~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Sep  7 10:33:52 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

Credit

Gandalf4a

Impact

This vulnerability allows a remote attacker to cause a denial of service on an affected gpac MP4Box. Exploiting this vulnerability requires user interaction, as the target must access a malicious page or open a malicious file.
Occurrences

filter_pck.c L108

SEGV in /gpac/src/filter_core/filter_pck.c:108:6 in gf_filter_pck_new_alloc_internal
poc

https://github.com/gandalf4a/crash_report/blob/main/gpac/MP4Box/poc3/segv_108
asan

[32m[Dasher] No template assigned, using $File$_dash$FS$$Number$
[0m[31m[PCMReframe] Missing audio ch, cannot parse
[0mAddressSanitizer:DEADLYSIGNAL
=================================================================
==1019581==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x7f6b6fcf09b1 bp 0x7ffc80a77bd0 sp 0x7ffc80a77ae0 T0)
==1019581==The signal is caused by a READ memory access.
==1019581==Hint: address points to the zero page.
    #0 0x7f6b6fcf09b1 in gf_filter_pck_new_alloc_internal /home/user/fuzzing_gpac/gpac/src/filter_core/filter_pck.c:108:6
    #1 0x7f6b70284641 in pcmreframe_process /home/user/fuzzing_gpac/gpac/src/filters/reframe_rawpcm.c:399:19
    #2 0x7f6b6fdafa33 in gf_filter_process_task /home/user/fuzzing_gpac/gpac/src/filter_core/filter.c:2971:7
    #3 0x7f6b6fd7d47b in gf_fs_thread_proc /home/user/fuzzing_gpac/gpac/src/filter_core/filter_session.c:2105:3
    #4 0x7f6b6fd7b5cf in gf_fs_run /home/user/fuzzing_gpac/gpac/src/filter_core/filter_session.c:2405:3
    #5 0x7f6b6f62ac6a in gf_dasher_process /home/user/fuzzing_gpac/gpac/src/media_tools/dash_segmenter.c:1236:6
    #6 0x558186b736dc in do_dash /home/user/fuzzing_gpac/gpac/applications/mp4box/mp4box.c:4831:15
    #7 0x558186b64b6e in mp4box_main /home/user/fuzzing_gpac/gpac/applications/mp4box/mp4box.c:6245:7
    #8 0x7f6b6e629d8f in __libc_start_call_main csu/../sysdeps/nptl/libc_start_call_main.h:58:16
    #9 0x7f6b6e629e3f in __libc_start_main csu/../csu/libc-start.c:392:3
    #10 0x558186a8cdd4 in _start (/home/user/fuzzing_gpac/gpac/bin/gcc/MP4Box+0x82dd4) (BuildId: 53333ca7bff59dd9a3d1b2821e7c5f3a9aac76b9)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /home/user/fuzzing_gpac/gpac/src/filter_core/filter_pck.c:108:6 in gf_filter_pck_new_alloc_internal
==1019581==ABORTING

filter_pid.c L9257

SEGV in /gpac/src/filter_core/filter_pid.c:9257:68 in gf_filter_pid_get_rfc_6381_codec_string
poc

https://github.com/gandalf4a/crash_report/blob/main/gpac/MP4Box/poc3/segv_9257
asan

[32m[iso file] Unknown top-level box type freN
[0m[32m[iso file] Unknown box type 00000004 in parent moov
[0m[31m[iso file] Read Box type 00000200 (0x00000200) at position 1213 has size 0 but is not at root/file level. Forbidden, skipping end of parent box !
[0m[33m[iso file] Box "moov" (start 351) has 703 extra bytes
[0m[32m[iso file] Unknown top-level box type s7Fyp
[0m[33m[iso file] Box "tfdt" (start 2056) has 84 extra bytes
[0m[31m[iso file] Read Box type 00000001 (0x00000001) at position 2156 has size 0 but is not at root/file level. Forbidden, skipping end of parent box !
[0m[33m[iso file] Box "traf" (start 2028) has 470 extra bytes
[0m[32m[iso file] Unknown box type sdBp in parent traf
[0m[32m[iso file] Unknown box type shgp in parent traf
[0m[33m[iso file] saio box without flags nor aux info type and no default scheme, ignoring
[0m[33m[iso file] senc box without tenc, assuming MS smooth+piff
[0m[31m[isobmf] cannot parse senc, missing IV/crypto state
[0m[33m[iso file] Box "uuid" (start 3475) has 58 extra bytes
[0m[33m[iso file] Box "uuid" (start 3565) has 58 extra bytes
[0m[31m[iso file] Read Box type 00000000 (0x00000000) at position 3655 has size 0 but is not at root/file level. Forbidden, skipping end of parent box !
[0m[33m[iso file] Box "trgr" (start 3467) has 225 extra bytes
[0m[32m[Dasher] No template assigned, using $File$_dash$FS$$Number$
[0m[31m[IsoMedia] Failed to fetch initial sample 1 for track 1
[0mAddressSanitizer:DEADLYSIGNAL
=================================================================
==1019207==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000008 (pc 0x7f7712b4d940 bp 0x000000000000 sp 0x7ffd2c720c30 T0)
==1019207==The signal is caused by a READ memory access.
==1019207==Hint: address points to the zero page.
    #0 0x7f7712b4d940 in gf_filter_pid_get_rfc_6381_codec_string /home/user/fuzzing_gpac/gpac/src/filter_core/filter_pid.c:9257:68
    #1 0x7f7712c614f1 in dasher_get_rfc_6381_codec_name /home/user/fuzzing_gpac/gpac/src/filters/dasher.c:1993:9
    #2 0x7f7712c614f1 in dasher_update_rep /home/user/fuzzing_gpac/gpac/src/filters/dasher.c:2385:3
    #3 0x7f7712c2a38f in dasher_setup_rep /home/user/fuzzing_gpac/gpac/src/filters/dasher.c:2418:2
    #4 0x7f7712c2a38f in dasher_setup_period /home/user/fuzzing_gpac/gpac/src/filters/dasher.c:6511:4
    #5 0x7f7712c5be25 in dasher_switch_period /home/user/fuzzing_gpac/gpac/src/filters/dasher.c:6478:9
    #6 0x7f7712c14d45 in dasher_process /home/user/fuzzing_gpac/gpac/src/filters/dasher.c:9683:6
    #7 0x7f7712bafa33 in gf_filter_process_task /home/user/fuzzing_gpac/gpac/src/filter_core/filter.c:2971:7
    #8 0x7f7712b7d47b in gf_fs_thread_proc /home/user/fuzzing_gpac/gpac/src/filter_core/filter_session.c:2105:3
    #9 0x7f7712b7b5cf in gf_fs_run /home/user/fuzzing_gpac/gpac/src/filter_core/filter_session.c:2405:3
    #10 0x7f771242ac6a in gf_dasher_process /home/user/fuzzing_gpac/gpac/src/media_tools/dash_segmenter.c:1236:6
    #11 0x55884948e6dc in do_dash /home/user/fuzzing_gpac/gpac/applications/mp4box/mp4box.c:4831:15
    #12 0x55884947fb6e in mp4box_main /home/user/fuzzing_gpac/gpac/applications/mp4box/mp4box.c:6245:7
    #13 0x7f7711429d8f in __libc_start_call_main csu/../sysdeps/nptl/libc_start_call_main.h:58:16
    #14 0x7f7711429e3f in __libc_start_main csu/../csu/libc-start.c:392:3
    #15 0x5588493a7dd4 in _start (/home/user/fuzzing_gpac/gpac/bin/gcc/MP4Box+0x82dd4) (BuildId: 53333ca7bff59dd9a3d1b2821e7c5f3a9aac76b9)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /home/user/fuzzing_gpac/gpac/src/filter_core/filter_pid.c:9257:68 in gf_filter_pid_get_rfc_6381_codec_string
==1019207==ABORTING

dash_client.c L10106

SEGV in gf_dash_group_get_audio_channels in /gpac/src/media_tools/dash_client.c:10106:11

asan

[32m[DASH] Setting up period start 0 duration 0 xlink none ID P1
[0m[33m[DASH] Cannot compute default segment duration
[0m[32m[DASH] AS#1 changed quality to bitrate 256 kbps - sample rate 8000 (playback speed 1)
[0m[32m[DASH] AS#2 changed quality to bitrate 800 kbps - Width 1280 Height 720 FPS 29/1 (playback speed 1)
[0m[32m[DASHDmx] setting up group 0 from /home/user/vul/MP4Box/crash2/id000457sig11src000410time401783180execs3815819ophavocrep2
[0m[31m[DASHDmx] group 1 error locating plugin for segment - mime type video/mp4 name /home/user/vul/MP4Box/crash2/v1_init_gpac.mp4: Requested URL is not valid or cannot be found
[0mAddressSanitizer:DEADLYSIGNAL
=================================================================
==1019179==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x7f5653047580 bp 0x00000000000a sp 0x7ffc8ecc6ba0 T0)
==1019179==The signal is caused by a READ memory access.
==1019179==Hint: address points to the zero page.
    #0 0x7f5653047580 in __GI_____strtol_l_internal stdlib/../stdlib/strtol_l.c:291:10
    #1 0x559fac822e25 in strtol (/home/user/fuzzing_gpac/gpac/bin/gcc/MP4Box+0xf0e25) (BuildId: 53333ca7bff59dd9a3d1b2821e7c5f3a9aac76b9)
    #2 0x7f5654173566 in atoi /usr/include/stdlib.h:364:16
    #3 0x7f5654173566 in gf_dash_group_get_audio_channels /home/user/fuzzing_gpac/gpac/src/media_tools/dash_client.c:10106:11
    #4 0x7f5654173566 in gf_dash_group_get_quality_info /home/user/fuzzing_gpac/gpac/src/media_tools/dash_client.c:10263:25
    #5 0x7f56548eb4d7 in dashdmx_declare_properties /home/user/fuzzing_gpac/gpac/src/filters/dmx_dash.c:1619:7
    #6 0x7f56548eb4d7 in dashdmx_configure_pid /home/user/fuzzing_gpac/gpac/src/filters/dmx_dash.c:2050:2
    #7 0x7f565470740c in gf_filter_pid_configure /home/user/fuzzing_gpac/gpac/src/filter_core/filter_pid.c:876:6
    #8 0x7f56547262a6 in gf_filter_pid_connect_task /home/user/fuzzing_gpac/gpac/src/filter_core/filter_pid.c:1230:3
    #9 0x7f565477d47b in gf_fs_thread_proc /home/user/fuzzing_gpac/gpac/src/filter_core/filter_session.c:2105:3
    #10 0x7f565477b5cf in gf_fs_run /home/user/fuzzing_gpac/gpac/src/filter_core/filter_session.c:2405:3
    #11 0x7f565402ac6a in gf_dasher_process /home/user/fuzzing_gpac/gpac/src/media_tools/dash_segmenter.c:1236:6
    #12 0x559fac89b6dc in do_dash /home/user/fuzzing_gpac/gpac/applications/mp4box/mp4box.c:4831:15
    #13 0x559fac88cb6e in mp4box_main /home/user/fuzzing_gpac/gpac/applications/mp4box/mp4box.c:6245:7
    #14 0x7f5653029d8f in __libc_start_call_main csu/../sysdeps/nptl/libc_start_call_main.h:58:16
    #15 0x7f5653029e3f in __libc_start_main csu/../csu/libc-start.c:392:3
    #16 0x559fac7b4dd4 in _start (/home/user/fuzzing_gpac/gpac/bin/gcc/MP4Box+0x82dd4) (BuildId: 53333ca7bff59dd9a3d1b2821e7c5f3a9aac76b9)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV stdlib/../stdlib/strtol_l.c:291:10 in __GI_____strtol_l_internal
==1019179==ABORTING
```

**Comment by aureliendavid** [comment]
part 1 was solved with #2632

part 2 solved by above commit

part 3 was solved with #2638

**Comment by rbouqueau** [comment]
It is ok to assign a CVE.
